### ISSUE TITLE
Keypad fails on leader map with Which-Key popup visible

### ISSUE THREAD

**Issue by 45mg** [meow-edit/meow#694]
After 8ae8b2c, pressing any key under the leader keymap (as defined in `meow-keymap-alist`, `mode-specific-map` by default) when the Which-Key popup is showing will result in '\<key\> is undefined'.

# Reproducing

1) Clone the meow repo, and add the following to a file `my-setup.el` in the repo directory:
```emacs-lisp
(require 'meow)
(setq package-user-dir "~/deleteme") ; don't mess up your existing package dir;
                                    ; remove this dir if running the test again
(define-key mode-specific-map "e" #'emacs-version)
(meow-global-mode)
```

2) Now run `emacs -q -L . -l my-setup.el`.

3) Try `SPC e`, it will always execute `emacs-version`, as expected.

4) Now, do `M-x package-install which-key RET`, `M-x which-key-mode RET`.

5) Press `SPC`, wait for which-key popup to show, then press `e`. You will get 'e is undefined' error.

6) Press `SPC e` quickly, without waiting for the popup to show. `emacs-version` is executed.

# Workaround(?!)

For some reason, setting `meow-keypad-leader-dispatch` to `"C-c"` seems to fix the issue. But I have no idea how or why this works or what the side effects of it might be.

# Possibly related

https://github.com/meow-edit/meow/issues/691 - note that same workaround seemed to work
https://github.com/meow-edit/meow/issues/686#issuecomment-2525261407 - again, same workaround

**Comment by DogLooksGood** [comment]
I can reproduce this issue.

**Comment by DogLooksGood** [comment]
I think it's because how keymap leader is handled. 

The leader can either be a keymap or a key sequences. When it's a keymap, we used to raise the priority of that keymap by setting it as `overriding-local-map`. It used to work because each keypad input is a individual command, now it conflicts with which-key. I just updated how it's handled. Now we lookup this keymap directly instead of raising its priority. 

Try the latest commit.
